boot/bootutil/src/boot_record.c

```diff
@@ -23,19 +24,19 @@
 
 #include "mcuboot_config/mcuboot_config.h"
 
+#if defined(__ZEPHYR__) && defined(MCUBOOT_DATA_SHARING)
+#include <app_version.h>
+#endif
+
 #if defined(MCUBOOT_MEASURED_BOOT) || defined(MCUBOOT_DATA_SHARING)
 #include "bootutil/boot_record.h"
 #include "bootutil/boot_status.h"
 #include "bootutil_priv.h"
 #include "bootutil/image.h"
+#include "../../bootutil/src/bootutil_priv.h"
 #include "flash_map_backend/flash_map_backend.h"
 
-/* Error codes for using the shared memory area. */
-#define SHARED_MEMORY_OK            (0)
-#define SHARED_MEMORY_OVERFLOW      (1)
-#define SHARED_MEMORY_OVERWRITE     (2)
-#define SHARED_MEMORY_GEN_ERROR     (3)
-
+#if !defined(MCUBOOT_CUSTOM_DATA_SHARING_FUNCTION)
```

Comment:
What is the mcuboot size implication of using a `CUSTOM_DATA_SHARING_FUNCTION` compared to the existing solution? Does this `CUSTOM_DATA_SHARING_FUNCTION` have any benefits ?

Reply:
The provided solution is cleaner and better than before, but it can still be avoided completely.